orttraining/orttraining/python/training/experimental/gradient_graph/_gradient_graph_tools.py

```diff
@@ -0,0 +1,93 @@
+from pathlib import Path
+from typing import Any, Callable, Optional, Union
+
+import torch
+from onnxruntime.capi._pybind_state import GradientGraphBuilder
+from torch.onnx import TrainingMode
+
+from ...ortmodule._custom_op_symbolic_registry import CustomOpSymbolicRegistry
+
+
+def export_gradient_graph(
+        model: torch.nn.Module,
+        loss_fn: Callable[[Any, Any], Any],
+        example_input: torch.Tensor,
+        example_labels: torch.Tensor,
+        gradient_graph_path: Union[Path, str], intermediate_graph_path: Optional[Union[Path, str]] = None,
+        opset_version=12) -> None:
+    r"""
+    Build a gradient graph for `model` so that you can output gradients in an inference session when given a specific input and labels.
+
+    Args:
+        model (torch.nn.Module): A gradient will be built for this model.
+
+        loss_fn (Callable[[Any, Any], Any]): A function to compute the loss given the model's output and the `example_labels`.
+
+        example_input (torch.Tensor): Example input that you would give your model for inference/prediction.
+
+        example_labels (torch.Tensor): The expected labels for `example_input`.
+        This could be the output of your model when given `example_input` but it might be different if your loss function expects labels to be different (e.g. when using CrossEntropyLoss).
+
+        gradient_graph_path (Union[Path, str]): The path to where you would like to save the gradient graph.
+
+        intermediate_graph_path (Optional[Union[Path, str]): The path to where you would like to save any intermediate graphs that are needed to make the gradient graph.
+        Defaults to `gradient_graph_path` and it will be overwritten if this function executes successfully.
+
+        opset_version (int): See `torch.onnx.export`.
+    """
+
+    # Is this needed?
```

Comment:
Is this needed?

Reply:
Good catch. It's not needed and I removed it. Sorry, there's still some things I'm checking.